project/src/main/java/com/google/sps/servlets/Exposure.java

```diff
@@ -0,0 +1,131 @@
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Query.CompositeFilter;
+import com.google.appengine.api.datastore.Query.CompositeFilterOperator;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.HashMap;
+ 
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+/** Servlet that posts and loads announcemnts*/
+@WebServlet("/exposure")
+public class Exposure extends HttpServlet {
+  DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  UserService userService = UserServiceFactory.getUserService();
+  GsonBuilder gsonBuilder = new GsonBuilder();
+  Gson gson = gsonBuilder.create();
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      String action = (String) request.getParameter("action");
+      HttpSession session = request.getSession(false); 
+      String email = (String) session.getAttribute("email");
+      Filter emailFilter =
+        new FilterPredicate("email", FilterOperator.EQUAL, email);
+      Query query = new Query("User").setFilter(emailFilter);
+      PreparedQuery results = datastore.prepare(query);
+      if (action.equals("revert")) {
+          for (Entity entity: results.asIterable()) {
+              entity.setProperty("diagnosed", "negative");
+              datastore.put(entity);
+              session.setAttribute("diagnosed", "negative");
+          }
+          return;
+      }
+      for (Entity entity: results.asIterable()) {
+          entity.setProperty("diagnosed", "positive");
+          datastore.put(entity);
+          session.setAttribute("diagnosed", "positive");
+      }
+
+      
+      query = new Query("LocationLog").setFilter(emailFilter);
+      results = datastore.prepare(query);
+      PreparedQuery all;
+      Map<String, String> emails = new HashMap<String, String>();
+      for (Entity entity : results.asIterable()) {
+          Filter dateFilter = 
+            new FilterPredicate("date", FilterOperator.EQUAL, entity.getProperty("date"));
+          Filter latFilter = 
+            new FilterPredicate("lat", FilterOperator.EQUAL, entity.getProperty("lat"));
```

Comment:
`lat` can you use fullword , same as `lng`

Reply:
Done.